### PR TITLE
Update isochrones_layer_proc.py

### PR DESCRIPTION
**Pull request by mattamach**
Add additional parameter for the "smoothing factor" [0, 100] to 'isochrones from layer' processing algorithm.

### CODE REVIEW

**Comment by mattamach** [comment]
This b.t.w adresses Issue [https://github.com/GIScience/orstools-qgis-plugin/issues/172]

**Comment by TheGreatRefrigerator** [comment]
Note:
- also needs to be done for the iso from point processing
- not set smoothing instead of defaulting to `0`

File: ORStools/proc/isochrones_layer_proc.py [hunk 1]
```diff
@@ -82,9 +86,22 @@ def __init__(self):
                 name=self.IN_RANGES,
                 description="Comma-separated ranges [min or m]",
                 defaultValue="5, 10"
-            )
+            ),
+            QgsProcessingParameterNumber(
+                name=self.IN_SMOOTHING,
+                description="Smoothing factor between 0 [fine grained] and 100 [concave hull]",
```
Comment (koebi) [inline ORStools/proc/isochrones_layer_proc.py:92]: Where does `[concave hull]` come from? Are you sure about this?

File: ORStools/proc/isochrones_layer_proc.py [hunk 2]
```diff
@@ -82,9 +86,22 @@ def __init__(self):
...
+                minValue=0,
+                maxValue=100,
+                optional=True
+            ),
+
         ]
 
+        # Set flag of smoothness parameters to advanced
+        smooth_param = next((i for i in self.PARAMETERS if i.name() == self.IN_SMOOTHING), None)
+        smooth_param.setFlags(smooth_param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
```
Comment (koebi) [inline ORStools/proc/isochrones_layer_proc.py:103]: All other advanced Params are set in https://github.com/GIScience/orstools-qgis-plugin/blob/4d9e4ad0e19a93637ac748174d0c20f47a578ea5/ORStools/proc/base_processing_algorithm.py#L209
with the info on what params are advanced defined in 
https://github.com/GIScience/orstools-qgis-plugin/blob/4d9e4ad0e19a93637ac748174d0c20f47a578ea5/ORStools/common/__init__.py#L52

If there's no good reason to deviate from this, I'd suggest moving this there :)

File: ORStools/proc/isochrones_layer_proc.py [hunk 3]
```diff
@@ -129,14 +148,20 @@ def processAlgorithm(self, parameters, context, feedback):
...
                 "range": ranges_proc,
                 "attributes": ['total_pop'],
                 "id": id_value,
-                "options": options
-            })
+                "options": options,
+            }
+
+            if smoothing or smoothing == 0:
+                params["smoothing"] = smoothing
```
Comment (koebi) [inline ORStools/proc/isochrones_layer_proc.py:161]: We're just checking for `None` here, right? I'd suggest the following for clarity then:
```suggestion
            # only include smoothing if set
            if smoothing is not None:
                params["smoothing"] = smoothing
```

File: ORStools/proc/isochrones_point_proc.py [hunk 4]
```diff
@@ -67,9 +70,21 @@ def __init__(self):
...
+                defaultValue=None,
+                minValue=0,
+                maxValue=100,
+                optional=True
+            ),
         ]
 
+        # Set flag of smoothness parameters to advanced
+        smooth_param = next((i for i in self.PARAMETERS if i.name() == self.IN_SMOOTHING), None)
+        smooth_param.setFlags(smooth_param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
```
Comment (koebi) [inline ORStools/proc/isochrones_point_proc.py:86]: see above.

File: ORStools/proc/isochrones_point_proc.py [hunk 5]
```diff
@@ -102,9 +118,12 @@ def processAlgorithm(self, parameters, context, feedback):
             "range": ranges_proc,
             "attributes": ['total_pop'],
             "id": None,
-            "options": options
+            "options": options,
         }
 
+        if smoothing or smoothing == 0:
+            params["smoothing"] = smoothing
+
```
Comment (koebi) [inline ORStools/proc/isochrones_point_proc.py:126]: As above:
```suggestion
            # only include smoothing if set
            if smoothing is not None:
                params["smoothing"] = smoothing
```

**Review by koebi — CHANGES_REQUESTED** [review]
Translation and Changelog.md-entry are missing.

File: ORStools/proc/isochrones_layer_proc.py (hunk 1 above)
Comment (Merydian) [reply ORStools/proc/isochrones_layer_proc.py:92]: Not sure as i didn't add this. Should we change it to `Smoothing factor between 0 [particularized] and 100 [generalized]`? The documentation in the API Playground doesn't state anything regarding concave hull.

Comment (koebi) [reply ORStools/proc/isochrones_layer_proc.py:92]: I'd propose `[detailed]` and `[generalized]`.

**Comment by koebi** [comment]
rebased in #199

**Pull request closed without merging**